### PR TITLE
Phalanx/Panzer: long term fix to view-of-views for kokkos serial backend changes

### PR DESCRIPTION
**Pull request by rppawlo**
@trilinos/phalanx 

## Motivation

Changes to the kokkos serial backend in https://github.com/kokkos/kokkos/pull/7080 was causing deadlock in the destructor of  a host-host view-of-views as reported in https://github.com/kokkos/kokkos/issues/7092. A short term fix that required users to call a function to free the inner host-host view was added in  https://github.com/trilinos/Trilinos/pull/13188. This commit cleans up the v-of-v and provides a long term fix so that users don't have to call the free function. Also did some cleanup on the naming and removed some older alternate experimental versions of v-of-v.

## Related Issues

* Closes `put-issue-number-here`

## Stakeholder Feedback

Fixed.

## Testing

All the v-of-v tests.

### CODE REVIEW

**Comment by rppawlo** [comment]
@sebrowne - everything passed on this PR except intel and nothing for that build was posted to cdash. Retesting now, but might need some help since I can't see the build error.

**Comment by rppawlo** [comment]
> @sebrowne - everything passed on this PR except intel and nothing for that build was posted to cdash. Retesting now, but might need some help since I can't see the build error.

@sebrowne - nevermind - just saw your email about the TPLs. Sorry for the spam...

File: packages/phalanx/src/design/README.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+The design directory contains some ideas and tests we should keep for
+informational purposes.
+
+The ViewOfView files show three different designs. The firsst two were
```
Comment (maartenarnst) [inline packages/phalanx/src/design/README.txt:4]:
```suggestion
The ViewOfView files show three different designs. The first two were
```

Comment (rppawlo) [reply packages/phalanx/src/design/README.txt]: done!

**Pull request merged**